source/SimpleServers/PeanutButter.SimpleHTTPServer/TcpClientExtensions.cs

```diff
@@ -20,6 +20,8 @@ public static string ReadLine(this TcpClient client)
             var stream = client.GetStream();
             stream.ReadTimeout = 3000;
             var data = new List<char>();
+            // Wait for a while for data coming to the stream
+            Thread.Sleep(10);
```

Comment:
`Thread.Sleep` here is a smell - what happens if something (eg machine load) causes whatever we're waiting for  to take longer than 10ms?

The prior patch also caused complete failure at my end - I'm reverting it in favor of this:

```csharp
  /// <summary>
        /// Reads one line of data from the TcpClient
        /// </summary>
        /// <param name="client"></param>
        /// <returns></returns>
        public static string ReadLine(this TcpClient client)
        {
            var stream = client.GetStream();
            if (!stream.CanRead)
            {
                throw new InvalidOperationException(
                    "Cannot read from stream"
                );
            }

            var data = new List<char>();
            var readFails = 0;
            do
            {
                var thisChar = stream.ReadByte();
                if (thisChar == '\n') break;
                if (thisChar == '\r') continue;
                if (thisChar < 0)
                {
                    if (++readFails > 10)
                    {
                        // it's possible there's just a hiccup in the stream?
                        break;
                    }

                    Thread.Sleep(0);
                    continue;
                }

                data.Add(Convert.ToChar(thisChar));
            } while (stream.DataAvailable);

            return string.Join(string.Empty, data);
        }
```
which still observes stream.DataAvailable as per the example at https://learn.microsoft.com/en-us/dotnet/api/system.net.sockets.networkstream.dataavailable?view=net-6.0#system-net-sockets-networkstream-dataavailable

According to the docs, ReadByte() should only return -1 at the end of the stream, but looking at the implementation, it looks like it could fail for any read error, so I'm giving it a few more spins. Note that `Thread.Sleep(0)` is a trick to allow other threads a chance to be scheduled - not an actual attempt to sleep.

Reply:
Are you sure about stream.ReadByte() before stream.DataAvailable?
Client may connect and disconnect immediately without sending any data.

Reply:
well, I see https://learn.microsoft.com/en-us/dotnet/api/system.net.sockets.networkstream.readbyte?view=net-6.0 suggests an `IOException`, so perhaps that should be catered for...

Reply:
@andreyserenkov try 3.0.38 - all ReadByte operations wrapped in a try/catch with failure returning -2; outer reader treats -2 as an epic fail, -1 as a possibly transitive one.